src/cursor/cur_index.c

```diff
@@ -347,6 +347,30 @@ __curindex_search_near(WT_CURSOR *cursor, int *exact)
     API_END_RET(session, ret);
 }
 
+/*
+ * __curindex_bound --
+ *     WT_CURSOR->bound method for the index cursor type.
+ *
+ */
+static int
+__curindex_bound(WT_CURSOR *cursor, const char *config)
+{
+    WT_CURSOR *child;
+    WT_CURSOR_INDEX *cindex;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    cindex = (WT_CURSOR_INDEX *)cursor;
+    JOINABLE_CURSOR_API_CALL(cursor, session, bound, NULL);
+
+    /* Grab the primary cursor and call bound function. */
+    cindex = (WT_CURSOR_INDEX *)cursor;
```

Comment:
I'm not sure if this is correct, is the child the index cursor itself?

Reply:
From my understanding, the child cursor refers to the index table because it opens on the underlying index table source here
`WT_ERR(__wt_open_cursor(session, idx->source, cursor, cfg, &cindex->child));`